newIDE/app/src/EditorFunctions/index.js

```diff
@@ -3891,6 +3978,103 @@ const changeScenePropertiesLayersEffects: EditorFunction = {
       });
     }
 
+    if (changed_groups) {
+      const groups = scene.getObjects().getObjectGroups();
+      changed_groups.forEach(changed_group => {
+        const groupName = SafeExtractor.extractStringProperty(
+          changed_group,
+          'group_name'
+        );
+        const newGroupName = SafeExtractor.extractStringProperty(
+          changed_group,
+          'new_group_name'
+        );
+        const deleteThisGroup = SafeExtractor.extractBooleanProperty(
+          changed_group,
+          'delete_this_group'
+        );
+        const objects = SafeExtractor.extractArrayProperty(
+          changed_group,
+          'objects'
+        );
+        if (groupName === null) {
+          warnings.push(
+            `Missing "group_name" in an item of \`changed_groups\`. It was ignored and not changed. Make sure you follow the exact format for changing group properties.`
+          );
+          return;
+        }
+
+        let foundGroup: gdObjectGroup;
+        if (!groups.has(groupName)) {
+          // Create the group if it does not exist yet.
+          foundGroup = groups.insertNew(groupName, groups.count());
+        } else {
+          foundGroup = groups.get(groupName);
+        }
+
+        if (deleteThisGroup) {
+          groups.remove(groupName);
+          changes.push(
+            `Deleted group "${groupName}" from scene "${scene_name}".`
+          );
+        } else {
+          if (newGroupName) {
+            gd.WholeProjectRefactorer.objectOrGroupRenamedInScene(
+              project,
+              scene,
+              foundGroup.getName(),
+              newGroupName,
+              /* isObjectGroup=*/ true
+            );
+            foundGroup.setName(newGroupName);
+            changes.push(
+              `Renamed group "${groupName}" to "${newGroupName}" in scene "${scene_name}".`
+            );
+          }
+          if (objects) {
+            const newObjectNames = objects
+              .map(object =>
+                SafeExtractor.extractStringProperty(object, 'object_name')
+              )
+              .filter(Boolean);
+            // Remove objects that are not in the list, and add new objects.
+            const currentObjectNames = foundGroup
+              .getAllObjectsNames()
+              .toJSArray();
+            currentObjectNames.forEach(objectName => {
+              if (!newObjectNames.includes(objectName)) {
+                foundGroup.removeObject(objectName);
+              }
+            });
+            const globalObjects = project.getObjects();
+            const sceneObjects = project.getLayout(scene_name).getObjects();
```

Comment:
You already have it?
```suggestion
            const sceneObjects = scene.getObjects();
```

I see any get as suspicious if I don't see a check before for existence.